include/mitsuba/core/distr_1d.h

```diff
@@ -472,7 +502,41 @@ template <typename Value> struct ContinuousDistribution {
     }
 
 private:
-    void compute_cdf(const ScalarFloat *pdf, size_t size) {
+    void compute_cdf() {
+        static_assert(dr::is_jit_v<Float>,
+                      "This method should only be used in JIT variants!");
+
+        if (m_pdf.size() < 2)
+            Throw("ContinuousDistribution: needs at least two entries!");
+        if (!(m_range.x() < m_range.y()))
+            Throw("ContinuousDistribution: invalid range!");
+        if (!dr::all(m_pdf >= 0.f))
+            Throw("DiscreteDistribution: entries must be non-negative!");
+        if (!dr::any(m_pdf > 0.f))
+            Throw("DiscreteDistribution: no probability mass found!");
+
+        uint32_t size = m_pdf.size() - 1;
+        m_interval_size_scalar = (m_range.y() - m_range.x()) / size;
+        auto index_1_to_n = dr::arange<dr::uint32_array_t<Float>>(1, size + 1);
+
+        m_cdf =
+            m_interval_size_scalar *
+            (dr::gather<Float>(dr::prefix_sum(m_pdf, false), index_1_to_n) -
```

Comment:
I didn't quite understand the logic here -- you're doing a prefix sum and then throwing away the first element. Isn't that what an `exclusive` prefix` sum does (with `true` second argument)